vespalib/src/vespa/vespalib/net/tls/maybe_tls_crypto_socket.cpp

```diff
@@ -77,12 +77,14 @@ class MyCryptoSocket : public CryptoSocket
         if (frame > 0) {
             memcpy(buf, src.data, frame);
             _buffer.evict(frame);
+            _buffer.drop_if_empty();
```

Comment:
should not this `drop_if_empty` be in `drop_empty_buffers()` below?

Reply:
Actually, no. This is to discard the TLS snoop buffer after it has been drained in the non-tls case. We do direct read/write to the socket in this mode, and do not need the 4_Ki buffer used for snooping during handshaking.